src/prometheus/deploy/prometheus-deployment.yaml.template

```diff
@@ -42,11 +42,13 @@ spec:
       hostNetwork: true
       initContainers:
       - name: init
-        image: bash:4
+        image: busybox:1.31.1
+        securityContext:
+          runAsUser: 0
         volumeMounts:
         - name: prometheus-data
           mountPath: /prometheus-data
-        command: ["chmod", "777", "/prometheus-data"] # newly create dir have permission 755, which makes prometheus container unable to write
+        command: ["chown", "-R", "nobody:nogroup", "/prometheus-data"] # prometheus run as nobody:nogroup, refer to: https://github.com/prometheus/prometheus/pull/2859
```

Comment:
do you have the permission to run "chown -R"?

Reply:
Sure please refer to:
https://github.com/microsoft/pai/blob/125fc65fb4dcb167888db7584e5e198d13006fd6/src/prometheus/deploy/prometheus-deployment.yaml.template#L46-L47
Init container will run as root user, and I tested in INT bed

Reply:
what if the path exists and it is readonly to the owner itself?

Reply:
If so, initContainer will failed, and k8s will try to restart the Prometheus according to restartPolicy.
Refer to: https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#detailed-behavior. Service will not start at the point.

Reply:
how about just add a write permission by default?

Reply:
It will also give Prometheus permission to modify previous data. And the problem will also solved. But 777 permission seems too general, and will leave prometheus data folder with files which own by different users (root and nobody).